bin/node/runtime/src/impls.rs

```diff
@@ -111,6 +111,29 @@ impl ProposalProvider<AccountId, Hash, RuntimeCall> for AllianceProposalProvider
 	}
 }
 
+pub struct Hasher;
+impl pallet_pov_limit::Hasher for Hasher {
+	type Out = ();
+
+	fn hash(_: &[u8]) -> Self::Out {
+		// this calculates factorial. should be replaced by a hashing algorithm
+		// but `Blake2Hasher` cannot be used here since this is no_std.
```

Comment:
Should we use a different hashing algorithm then?

Reply:
Just use `T::Hash` from the `frame_system`. The exact hash does not matter as long as it has consistent weight.